pkg/virt-controller/watch/migration.go

```diff
@@ -515,6 +532,15 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 	return nil
 }
 
+func (c *MigrationController) setMigrationFailed(migration *virtv1.VirtualMachineInstanceMigration) error {
+	migration.Status.Phase = virtv1.MigrationFailed
+	if _, exists := migration.GetAnnotations()[virtv1.EvacuationMigrationAnnotation]; !exists {
+		return nil
+	}
+
```

Comment:
Why do we need this if statement?

Reply:
It is just because the `descheduler.alpha.kubernetes.io/eviction-in-progress` annotation is only added for evacuation migrations. There is no need to check the source virt-launcher pod annotations if the migration isn't an evacuation migration.
I am ready to drop this `if` you think that it does not worth it, but IMO this would save execution time and errors coming from reading the pod cache.

Reply:
I understand, but this code doesn't involve an API call, so the early exit won't notably improve performance. Therefore, I believe having less complexity is preferred in this case.

Reply:
I understand, but this code doesn't involve an API call, so the early exit won't notably improve performance. Therefore, I believe having less complexity is preferred in this case.

Reply:
Done